### PR TITLE
[QA-424]: Split OIDC and Auth calls in Auth scenarios

### PR DESCRIPTION
**Pull request by abhinaysrivastavaperf**
### What?
Split OIDC and Auth calls in `sign-up` and `sign-in` scenarios

#### Changes:
- Split OIDC and Auth calls in `sign-up` and `sign-in` scenarios.
- Modified tag names.
- Corrected the position of the `iterationCompleted` metric 

---

### Why?
To evaluate the performance metrics of both Orchestration and Authentication calls separately.

### CODE REVIEW

File: deploy/scripts/src/authentication/test.ts [hunk 1]
```diff
@@ -265,118 +273,180 @@ export function signUp (): void {
...
   // 25% of users logout
   if (Math.random() <= 0.25) {
     sleep(1)
-
     res = group('B01_SignUp_12_Logout', () =>
       timeRequest(() => res.submitForm({
         params: { tags: { name: 'B01_SignUp_12_Logout' } }
       }), { isStatusCode200, ...pageContentCheck('Successfully signed out') }))
   }
-  iterationsCompleted.add(1)
```
Comment (Tom-Dann) [inline deploy/scripts/src/authentication/test.ts:282]: Keep this at the end of the function

File: deploy/scripts/src/authentication/test.ts [hunk 2]
```diff
@@ -265,118 +273,180 @@ export function signUp (): void {
...
+        { isStatusCode302 })
+        res = timeRequest(() => http.get(res.headers.Location, {
+          redirects: 0,
+          tags: { name: 'B02_SignIn_07_SMSMFA_EnterOTP_02_OIDCCall' }
+        }),
+        { isStatusCode302 })
+
+        rpStubCheck = (res.headers.Location).includes('auth-stub')
+        acceptNewTerms = (res.headers.Location).includes('signin')
+
```
Comment (Tom-Dann) [inline deploy/scripts/src/authentication/test.ts:415]: Is this check sufficient to determine if the new terms and conditions screen is presented?

File: deploy/scripts/src/authentication/test.ts [hunk 3]
```diff
@@ -265,118 +273,180 @@ export function signUp (): void {
...
     sleep(1)
 
-    res = group('B01_SignIn_10_Logout POST', () =>
+    res = group('B02_SignIn_09_Logout POST', () =>
       timeRequest(() => res.submitForm({
-        params: { tags: { name: 'B01_SignIn_10_Logout' } }
+        params: { tags: { name: 'B02_SignIn_09_Logout' } }
       }), { isStatusCode200, ...pageContentCheck('Successfully signed out') }))
   }
-  iterationsCompleted.add(1)
```
Comment (Tom-Dann) [inline deploy/scripts/src/authentication/test.ts:381]: Keep this at the end of the function

File: deploy/scripts/src/authentication/test.ts [hunk 4]
```diff
@@ -265,118 +273,180 @@ export function signUp (): void {
...
+        }),
+        { isStatusCode302 })
+        res = timeRequest(() => http.get(res.headers.Location, {
+          redirects: 0,
+          tags: { name: 'B02_SignIn_05_AuthMFA_EnterTOTP_02_OIDCCall' }
+        }),
+        { isStatusCode302 })
+
+        rpStubCheck = (res.headers.Location).includes('auth-stub')
+        acceptNewTerms = (res.headers.Location).includes('signin')
```
Comment (Tom-Dann) [inline deploy/scripts/src/authentication/test.ts:372]:
```suggestion
        rpStubCheck = res.headers.Location.includes('auth-stub')
        acceptNewTerms = res.headers.Location.includes('signin')
```

**Review by Tom-Dann — CHANGES_REQUESTED** [review]

File: deploy/scripts/src/authentication/test.ts (hunk 2 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/authentication/test.ts:415]: Yes this check should be sufficient as this to checking the URL of the redirect. However, I have updated this to a slightly better check.

File: deploy/scripts/src/authentication/test.ts (hunk 4 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/authentication/test.ts:372]: Done

File: deploy/scripts/src/authentication/test.ts (hunk 1 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/authentication/test.ts:282]: The `logout` step is purely a stub call so this metric has been moved before that step to provide an accurate view of the number of `sign-ups` and `sign-ins`.

File: deploy/scripts/src/authentication/test.ts (hunk 3 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/authentication/test.ts:381]: Same as previous comment - The logout step is purely a stub call so this metric has been moved before that step to provide an accurate view of the number of sign-ups and sign-ins.

File: deploy/scripts/src/authentication/test.ts (hunk 1 above)
Comment (Tom-Dann) [reply deploy/scripts/src/authentication/test.ts:282]: If that last step is a stub call just remove it from the script entirely. This metric increment should be the last step in the function regardless

Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/authentication/test.ts:282]: Slight correction there. The logout step has a redirect call to Orchestrator. I have updated this to split the stub and OIDC calls in the logout step.
Also, the iterationCompleted metric has been moved at the end of the function.

File: deploy/scripts/src/authentication/test.ts (hunk 3 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/authentication/test.ts:381]: The logout step has a redirect call to Orchestrator. I have updated this to split the stub and OIDC calls in the logout step.
Also, the iterationCompleted metric has been moved at the end of the function.

**Review by Tom-Dann — APPROVED** [review]

**Pull request merged**